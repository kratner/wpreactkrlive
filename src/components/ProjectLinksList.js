import React, { Component } from 'react';
import { qryPostsFromLinksCategory } from './GetDataWPRESTAPI';

class ProjectLinksList extends Component {
  state = {
    posts: []
  };

  componentDidMount() {
    let appUrl = `${qryPostsFromLinksCategory}`;
    fetch(appUrl)
      .then(data => data.json())
      .then(data => {
        this.setState({
          posts: data
        });
      });
  }

  render() {
    let listProjectLinks = this.state.posts
      .filter(node => node.acf.type === 'project')
      .sort((a, b) =>
        a.acf.weight > b.acf.weight ? 1 : -1
      )
      .map((node, index) => {
        return (
          <li key={index}>
            <a
              href={node.acf.href}
              className={node.acf.class}
              title={node.title.rendered}
              target={node.acf.target}
            >
              {node.acf.text}
              <span
                className={
                  'icon-' + node.acf.icon + ' inline-icon'
                }
              />
            </a>
          </li>
        );
      });

    return (
      <div id="project-links">
        <div className="link-padding">
          <ul>{listProjectLinks}</ul>
        </div>
      </div>
    );
  }
}

export default ProjectLinksList;
