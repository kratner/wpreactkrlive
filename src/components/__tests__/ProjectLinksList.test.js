import React from "react";
import { shallow } from "enzyme";
import ProjectLinksList from "../ProjectLinksList";

describe("ProjectLinksList", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ProjectLinksList />);
    expect(wrapper).toMatchSnapshot();
  });
});
