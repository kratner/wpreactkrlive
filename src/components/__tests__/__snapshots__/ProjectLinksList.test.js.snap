// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectLinksList renders correctly 1`] = `ShallowWrapper {}`;
